GliaWidgets/Public/Glia/Glia.swift

```diff
@@ -113,22 +117,46 @@ public class Glia {
         }
 
         interactor?.addObserver(self) { [weak self] event in
-            guard let engagement = self?.environment.coreSdk.getCurrentEngagement(),
-                  engagement.source == .callVisualizer,
-                  case .screenShareOffer(let answer) = event else {
+            guard let engagement = self?.environment.coreSdk.getCurrentEngagement(), engagement.source == .callVisualizer else {
                 return
             }
 
-            self?.environment.coreSdk.requestEngagedOperator { operators, _ in
-                self?.callVisualizer.offerScreenShare(
-                    from: operators ?? [],
-                    configuration: Theme().alertConfiguration.screenShareOffer,
-                    accepted: {
-                        answer(true)
-                    }, declined: {
-                        answer(false)
-                    }
-                )
+            switch event {
+            case .screenShareOffer(answer: let answer):
+                self?.environment.coreSdk.requestEngagedOperator { operators, _ in
+                    self?.callVisualizer.offerScreenShare(
+                        from: operators ?? [],
+                        configuration: Theme().alertConfiguration.screenShareOffer,
+                        accepted: {
+                            answer(true)
+                        }, declined: {
+                            answer(false)
+                        }
+                    )
+                }
+            case let .upgradeOffer(offer, answer):
+                self?.environment.coreSdk.requestEngagedOperator { operators, _ in
+                    self?.callVisualizer.offerMediaUpgrade(
+                        from: operators ?? [],
+                        offer: offer,
+                        answer: answer,
+                        accepted: {
+                            answer(true, nil)
+                            self?.callVisualizer.coordinator?.showVideoCallViewController()
```

Comment:
I guess we need to have intermediate `showVideoCallViewController()` method in `CallVisualizer` and have `private` access level for `CallVisualizer.coordinator`.

Reply:
Why is that essential? Because then we also need to handle all the events as well because most of them will call a method from the coordinator.

Reply:
The idea is to work only through `CallVisualizer`

Reply:
@igorkravchenko what do you think?

Reply:
@rasmustautsglia  I guess @EgorovEI is taking about the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) principle. So accessing inner component like coordinator is implementation detail and should be hidden behind dedicated interface.

Reply:
I made coordinator private and made access to coordinator within callVisualizer [b362463](https://github.com/salemove/ios-sdk-widgets/pull/494/commits/b362463a70614c86a26c4f175133bd3d044ead9a)